tree/master/cloud/src/solution/Microsoft.Legal.MatterCenter.Service/Microsoft.Legal.MatterCenter.Repository/SharePointInfo/Taxonomy.cs

```diff
@@ -235,6 +235,7 @@ private TaxonomyResponseVM GetTaxonomyHierarchy (TermStore termStore, TermStoreD
         /// <returns>Serialized Object of Term Set</returns>
         private string GetManagedTermSetHierarchy(ClientContext clientContext, TermSet termSet, TermStoreDetails termStoreDetails)
         {
+            level = 3;
```

Comment:
We have to re-initialize the level when ever we need to get fresh data. If we do not do this, this value will go on incrementing